### PR TITLE
feat: Function.funext_iff and Sum.elim_eq_iff

### PR DESCRIPTION
**Pull request by madvorak**
I didn't really know where to put `Function.funext_iff` because `Std/Data/Sum/Lemmas` imports `Std/Data/Sum/Basic` and nothing else. Please refactor!

### CODE REVIEW

**Comment by madvorak** [comment]
awaiting-review

File: Std/Data/Sum/Lemmas.lean [hunk 1]
```diff
@@ -12,6 +12,9 @@ import Std.Data.Sum.Basic
 Theorems about the definitions introduced in `Std.Data.Sum.Basic`.
 -/
 
+theorem Function.funext_iff {β : α → _} {f₁ f₂ : ∀ x : α, β x} : f₁ = f₂ ↔ ∀ a, f₁ a = f₂ a :=
```
Comment (semorrison) [inline Std/Data/Sum/Lemmas.lean:15]:
```suggestion
theorem Function.funext_iff {β : α → Sort u} {f₁ f₂ : ∀ x : α, β x} : f₁ = f₂ ↔ ∀ a, f₁ a = f₂ a :=
```

Comment (semorrison) [reply Std/Data/Sum/Lemmas.lean:15]: Please move to `Std.Logic`, after `funext₃`.

**Comment by madvorak** [comment]
I get this error:
```
[109/165] Building Std.Dataerror: > c:\Users\mdvorak\.elan\toolchains\leanprover--lean4---v4.7.0-rc2\bin\leanc.exe -o .\.lake\build\bin\runLinter.exe .\.lake\build\ir\runLinter.c.o .\.lake\build\ir\Std\Tactic\Lint\Basic.c.o .\.lake\build\ir\Std\Tactic\Lint\Misc.c.o .\.lake\build\ir\Std\Tactic\OpenPrivate.c.o .\.lake\build\ir\Std\Util\LibraryNote.c.o .\.lake\build\ir\Std\Tactic\Lint\Simp.c.o .\.lake\build\ir\Std\Tactic\Lint\TypeClass.c.o .\.lake\build\ir\Std\Tactic\Lint\Frontend.c.o .\.lake\build\ir\Std\Tactic\Lint.c.o .\.lake\build\ir\Std\Data\List\Init\Attach.c.o .\.lake\build\ir\Std\Data\Array\Basic.c.o .\.lake\build\ir\Std\Lean\Util\Path.c.o
error: stderr:
ld.lld: error: too many exported symbols (got 65747, max 65535)
clang: error: linker command failed with exit code 1 (use -v to see invocation)
error: external command `c:\Users\mdvorak\.elan\toolchains\leanprover--lean4---v4.7.0-rc2\bin\leanc.exe` exited with code 1
```

**Comment by madvorak** [comment]
awaiting-review

File: Std/Logic.lean [hunk 2]
```diff
@@ -59,6 +59,9 @@ theorem funext₃ {β : α → Sort _} {γ : ∀ a, β a → Sort _} {δ : ∀ a
     {f g : ∀ a b c, δ a b c} (h : ∀ a b c, f a b c = g a b c) : f = g :=
   funext fun _ => funext₂ <| h _
 
+theorem Function.funext_iff {β : α → Sort u} {f₁ f₂ : ∀ x : α, β x} : f₁ = f₂ ↔ ∀ a, f₁ a = f₂ a :=
+  Iff.intro (fun hf _ => hf ▸ rfl) funext
```
Comment (fgdorais) [inline Std/Logic.lean:63]:
```suggestion
  ⟨congr_fun, funext⟩
```

**Review by fgdorais — CHANGES_REQUESTED** [review]

**Review by fgdorais — APPROVED** [review]

**Pull request merged**